### PR TITLE
[backport-2.8.x] fix(dns): fix the UDP socket leaking issue

### PR DESCRIPTION
**Pull request by liverpool8056**
cherry pick from https://github.com/Kong/kong/pull/10691

### Checklist

- [x] The Pull Request has tests
- [x] A changelog file has been created under `changelog/unreleased/kong` or `skip-changelog` label added on PR if changelog is unnecessary. [README.md](https://github.com/Kong/gateway-changelog/README.md)
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Issue reference

FTI-5803

### CODE REVIEW

**Review by oowl — APPROVED** [review]

File: kong/resty/dns/client.lua [hunk 1]
```diff
@@ -714,6 +713,13 @@ local function individualQuery(qname, r_opts, try_list)
 
   local result
   result, err = r:query(qname, r_opts)
+  -- Manually destroy the resolver.
+  -- When resovler is initialized, some socket resources are also created inside
+  -- resolver. As the resolver is created in timer-ng, the socket resources are
```
Comment (ADD-SP) [inline kong/resty/dns/client.lua:718]: There is no timer-ng in 2.8.x.

**Review by ADD-SP — CHANGES_REQUESTED** [review]

File: kong/resty/dns/client.lua (hunk 1 above)
Comment (liverpool8056) [reply kong/resty/dns/client.lua]: good catch

Comment (chobits) [reply kong/resty/dns/client.lua]: I think it has been resolved in https://github.com/Kong/kong/pull/12698/commits/81e19b2ffd4ec939db15291723e8931b7d1b5449, so resolve this conversation

**Review by chobits — APPROVED** [review]

File: CHANGELOG/unreleased/kong/udp-socket-recycle.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+message: "fix a bug causing udp socket leak"
```
Comment (ADD-SP) [inline CHANGELOG/unreleased/kong/udp-socket-recycle.yml:1]:
``` suggestion
message: "Fixed the UDP socket leaking issue of the DNS module."
```

**Review by ADD-SP — CHANGES_REQUESTED** [review]

**Review by ADD-SP — APPROVED** [review]

**Pull request merged**